js/foundation.magellan.js

```diff
@@ -93,15 +93,28 @@ class Magellan extends Plugin {
       _this._updateActive();
     });
 
-    this.$element.on({
-      'resizeme.zf.trigger': this.reflow.bind(this),
-      'scrollme.zf.trigger': this._updateActive.bind(this)
-    }).on('click.zf.magellan', 'a[href^="#"]', function(e) {
+    if (document.readyState === "complete") {
+      _this.$element.on({
+        'resizeme.zf.trigger': _this.reflow.bind(_this),
+        'scrollme.zf.trigger': _this._updateActive.bind(_this)
+      }).on('click.zf.magellan', 'a[href^="#"]', function(e) {
         e.preventDefault();
-        var arrival   = this.getAttribute('href');
+        var arrival   = _this.getAttribute('href');
         _this.scrollToLoc(arrival);
       });
-
+    } else {
+      $(window).one('load', function(){
+        _this.$element.on({
+          'resizeme.zf.trigger': _this.reflow.bind(_this),
+          'scrollme.zf.trigger': _this._updateActive.bind(_this)
+        }).on('click.zf.magellan', 'a[href^="#"]', function(e) {
+          e.preventDefault();
+          var arrival   = _this.getAttribute('href');
+          _this.scrollToLoc(arrival);
+        });
+      });
+    }
```

Comment:
Duplicated content.

We could do something like:
```js
// core.util.js
function onLoad(cb) {
  (document.readyState === "complete")
    ? cb()
    : $(window).one(cb);
 }
```
```js
// foundation.magellan.js
onLoad(() => {
  // code here
});
```

Reply:
Some util method for all components? Because we have this duplicated part probably in more components and tests where we do the same check.

Reply:
Sounds good. A candidate for another PR for the core utils?

Reply:
Ok, let's merge this as this, I will refactor all that in https://github.com/zurb/foundation-sites/pull/10947.

Reply:
Makes sense 👍 That should ensure that each PR produces consistent behavior and better overview. 

Personally I am not a big fan of mixing too much and big changes in one PR and separate them. But this is just my opinion.

Reply:
One PR = one action, with little or big changes. We can do a PR across the whole framework to refactor events on initialization, it's not a problem this we do more or less the same thing in all components and we're not lost in changes.